src/main/java/seedu/recipe/model/recipe/ingredient/IngredientQuantity.java

```diff
@@ -1,35 +1,49 @@
 package seedu.recipe.model.recipe.ingredient;
 
-import java.util.Objects;
-
 import seedu.recipe.model.recipe.exceptions.RecipeQuantityInvalidArgumentException;
-import seedu.recipe.model.recipe.unit.RecipeIngredientQuantityUnit;
 
 /**
  * Represents the Quantity field associated with a {@code IngredientBuilder}.
  */
 public class IngredientQuantity {
     public static final String MESSAGE_CONSTRAINTS =
-            "The quantity for a Recipe IngredientBuilder should consist of this format:\n"
-            + "`{amount} {unit}`, and the unit should comprise of alphabetic characters, "
-            + "where the amount can either be a non-zero decimal, `A/a`, or `One/one`."
+            "The quantity field for a Recipe Ingredient should consist of this format: "
+            + "`{amount} {unit}`, where the amount can either be a non-zero decimal, i.e. `A/a` or `One/one`, "
+            + "and the unit should comprise of alphabetic characters, with minimal use of "
+            + "trailing periods ('.') and hyphens."
             + "i.e. `1 gram`, `1.5 L`, `A pinch of`, `One oz.`";
 
     private static final String ALPHA_AMOUNT_REGEX = "[aA]|[Oo]ne";
+    private static final String RANGE_REGEX = "\\d+\\s*(\\-|to)\\s*\\d+";
     private static final String AMOUNT_REGEX = String.format(
-            "([1-9][0-9]*([\\./][0-9]*[1-9])?|%s)", ALPHA_AMOUNT_REGEX);
-    private static final String VALIDATION_REGEX = String.format("^%s(\\s+\\S+)+", AMOUNT_REGEX);
+            "([1-9][0-9]*|[0-9]+[\\./][0-9]*[1-9]|%s|%s)", ALPHA_AMOUNT_REGEX, RANGE_REGEX);
+    private static final String VALIDATION_REGEX = String.format("^%s(\\s+[A-Za-z]+\\.?)*", AMOUNT_REGEX);
 
-    private final double amount;
-    private final RecipeIngredientQuantityUnit unit;
+    private final String amount;
 
-    private IngredientQuantity(double amount, RecipeIngredientQuantityUnit unit) {
+    private IngredientQuantity(String amount) {
         this.amount = amount;
-        this.unit = unit;
     }
 
-    private static boolean isValidRecipeQuantity(String test) {
-        return test.matches(VALIDATION_REGEX);
+    private static boolean isValidRecipeQuantity(String candidate) {
+        assert candidate != null;
```

Comment:
Javadocs could be added because despite the method being named in a boolean way, candidate and the logic behind it are not immediately obvious